### ISSUE TITLE
Doesn't work in playground due to zlib code

### ISSUE THREAD

**Issue by e28eta** [tidwall/SwiftWebSocket#50]
When I include WebSocket.swift in a playground's sources directory, the playground fails to execute.

I'm able to work around the failure by commenting out the @_silgen_name declarations for libz methods and the uses of those methods in the Inflator/Deflator classes.

I'm not sure what the real solution to the problem is. Maybe using Apple's libcompression API, but I haven't (yet) taken the time to compare it to libz.

**Comment by tidwall** [comment]
I'm not familiar with playgrounds, but I did take a look and I couldn't find a solution just yet. It appears that there is no `#if !PLAYGROUND #endif` sort of preprocessor for the playground or swift. Also I can't find a way to include zlib into a playground project. 

I looked at other libraries that [Apple recommends](https://developer.apple.com/library/ios/documentation/Performance/Reference/Compression/) such as LZ4, LZMA, and LZFSE. All of them seem to require a linking stage which I don't think is possible in playground.

As it stands the only reasonable solution that I know of is to comment out the code. 

I'll keep looking and hopefully I'll find something.

**Comment by e28eta** [comment]
The documentation you linked to is what I was looking at. It looks trivially easy to use in the playground, just `import Compression` module. I didn't actually check, but I'm sure that's all you have to do to use them for any other project.

**Comment by tidwall** [comment]
Thanks for pointing that out. 

I was able to swap out the old code with the new and it seems to compile (though I haven't test if the inflating actually works). I had to include the Compression framework to the project. 

I'm still not able to get it to work in the playground. I added the WebSocket.swift and the libcompression.tbd file to the project.

![image](https://cloud.githubusercontent.com/assets/1156077/17159493/8b1685d2-5352-11e6-9455-d003aea8671c.png)

Here's a gist of the [WebSocket.swift](https://gist.github.com/tidwall/475bd2dd0a37d8195d7830ed0ef65e6a)

I must be close. Do you have any ideas?

**Comment by e28eta** [comment]
I'm having trouble with Xcode 7.3/Swift 2.2. It's choking on the libcompression code when it's in the Sources directory. It looks like it's not linking against the library correctly. I set up a minimal test case playground: a single file in Sources containing just enough code to call `compression_stream_init`. It's happy with references to the compression_stream struct, and the enum values for status. As soon as I add the method call, compilation of the Sources seems to fail:

```
dyld: warning: could not load inserted library '.../TestPlayground_Sources.framework/TestPlayground_Sources' because image not found
Playground execution failed: .../playground4.swift:18:8: error: no such module 'TestPlayground_Sources'
import TestPlayground_Sources
```

Taking the exact same playground & code (changing 1 line for Swift 3), it seems to be working in Xcode 8b2. Moving that minimal code out of the file in Sources directory and putting it into the playground's Contents.swift (the "playground" file), it works fine under 7.3. So maybe there's a 7.3 incompatibility or bug?

I did not need to include libcompression.tbd (and don't know what that does, but it seemed to make things worse in 7.3).

I may keep looking at this today or tomorrow, and see if I can get libcompression working with the Swift 3.0 version of WebSocket.swift that I've been playing with.
